allennlp/models/encoder_decoders/wikitables_semantic_parser.py

```diff
@@ -450,25 +448,28 @@ def _get_linking_probabilities(worlds: List[WikiTablesWorld],
                                                  dim=1,
                                                  index=Variable(linking_scores.data.new(row_type_index)).long())
 
-            # (num_question_tokens, num_entities_per_type)
+            # (num_question_tokens, num_entities_per_type + 1)
             cell_type_scores[:, 0] = 0
             row_type_scores[:, 0] = 0
 
-            probabilities_cell = torch.nn.functional.softmax(cell_type_scores)
-            probabilities_row = torch.nn.functional.softmax(row_type_scores)
+            probabilities_cell = torch.nn.functional.softmax(cell_type_scores, dim=1)
+            probabilities_row = torch.nn.functional.softmax(row_type_scores, dim=1)
```

Comment:
Looks like these softmaxes need to be masked.

Reply:
They don't, actually, because we constructed a `cell_type_scores` matrix per batch instance, with no padding.

Reply:
Sounds good.